clinica/iotools/converters/ixi_to_bids/ixi_to_bids_utils.py

```diff
@@ -0,0 +1,209 @@
+        subjects_to_filter = subjs_list_path.read_text().splitlines()
+    else:
+        subjects_to_filter = get_subjects_list_from_data(data_directory)
+    return filter_subjects_list(subjects_to_filter, clinical_data)
+
+
+def read_ixi_clinical_data(clinical_data_path: Path) -> pd.DataFrame:
+    """Reads and formats IXI clinical data"""
+    # todo : what keys ?
+    clinical_data = pd.read_excel(clinical_data_path / "IXI.xls")
+    clinical_data.dropna(inplace=True)
+    clinical_data.drop("DATE_AVAILABLE", axis=1, inplace=True)
+    clinical_data.rename(lambda x: x.lower(), axis=1, inplace=True)
+    clinical_data["ixi_id"] = clinical_data.ixi_id.apply(
+        lambda x: "IXI" + "0" * (3 - len(str(x))) + str(x)
+    )
+    clinical_data["session_id"] = "ses-M000"
+    return clinical_data
+
+
+def rename_ixi_modalities(input_mod: str) -> str:
+    if input_mod == "T1":
+        return "T1w"
+    elif input_mod == "T2":
+        return "T2w"
+    elif input_mod == "MRA":
+        return "angio"
+    elif input_mod == "PD":
+        return "PDw"
+    elif input_mod == "DTI":
+        return "dti"
+    else:
+        raise ValueError(
+            f"The modality {input_mod} is not recognized in the IXI dataset."
+        )
+
+
+def define_magnetic_field(hospital: str) -> str:
+    if hospital == "Guys" or hospital == "IOP":
+        return "1.5"
+    if hospital == "HH":
+        return "3"
+
+
+def get_img_data_df(data_directory: Path) -> pd.DataFrame:
+    """Finds paths for all images that are not DTI data and processes the info contained in their names"""
+    df = pd.DataFrame(
+        {
+            "img_path": [
+                path
+                for path in data_directory.rglob(pattern="IXI*.nii.gz")
+                if re.search(r"IXI\d{3}(-\w*){3}.nii.gz$", str(path))
+            ]
+        }
+    )
+    df = (
+        df.assign(img_name=lambda df: df.img_path.apply(lambda x: x.name))
+        .assign(img_name_no_ext=lambda df: df.img_name.apply(lambda x: x.split(".")[0]))
+        .assign(subject=lambda df: df.img_name_no_ext.apply(lambda x: x.split("-")[0]))
+        .assign(
+            bids_id=lambda df: df.subject.apply(
+                lambda x: bids_id_factory(StudyName.IXI).from_original_study_id(x)
+            )
+        )
+        .assign(hospital=lambda df: df.img_name_no_ext.apply(lambda x: x.split("-")[1]))
+        .assign(
+            modality=lambda df: df.img_name_no_ext.apply(
+                lambda x: rename_ixi_modalities(x.split("-")[3])
+            )
+        )
+        .assign(field=lambda df: df.hospital.apply(lambda x: define_magnetic_field(x)))
+        .assign(session="ses-M000")
+    )
+    return df
+
+
+def select_subject_data(data_df: pd.DataFrame, subject: str) -> pd.DataFrame:
+    """Select subject images to copy based on IXI id"""
+    return data_df[data_df["subject"] == subject]
+
+
+def bids_filename_from_ixi(img: pd.Series) -> str:
```

Comment:
Maybe something like `get_bids_filename_from_ixi` to start with an action verb.

Also, what is 'ixi' ? 

I would vote for something more explicit like `get_bids_filename_from_image_data` or `get_bids_filename_from_subject_row`... WDYT ?